src/controller/CHIPDeviceController.cpp

```diff
@@ -1171,7 +1168,13 @@ CHIP_ERROR DeviceCommissioner::ProcessOpCSR(const ByteSpan & CSR, const ByteSpan
 
     Device * device = &mActiveDevices[mDeviceBeingPaired];
 
-    // TODO: Validate CSR Nonce and signature
+    // TODO: Verify the OpCSR signature using pubkey from DAC
+    //       This will be done when device attestation is implemented.
+
+    // Verify that Nonce matches with what we sent
+    const ByteSpan nonce = device->GetCSRNonce();
+    VerifyOrReturnError(CSRNonce.size() == nonce.size(), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(memcmp(CSRNonce.data(), nonce.data(), CSRNonce.size()) == 0, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Could use the bits from #7444 once those land...  Nothing to do for now, though.